docs/tutorial/Tutorial.md

```diff
@@ -0,0 +1,981 @@
+by GC work units) using a single thread.
+
+*live*: An object that can still be accessed by other objects is live/alive.
+
+*mutator*: Something that 'mutates', or changes, the objects stored in memory. 
+That is to say, this is a running program.
+
+*plan*: A garbage collection algorithm composed of components from the MMTk.
+
+*policy*: A definition of the semantics and behaviour of a memory region. 
+Memory spaces are instances of policies.
+
+*scheduler*: Dynamically dispatches units of GC work to workers.
+
+*zeroing*, *zero initialization*: Initializing and resetting unused memory 
+bits to have a value of 0, generally to improve memory safety.
+
+See also: [Further Reading](#further-reading)
+
+[**Back to table of contents**](#contents)
+***
+## Preliminaries
+### Set up MMTk and OpenJDK
+#### Basic set up
+This tutorial can be completed with any binding. However, for the sake of 
+simplicity, only the setup for the OpenJDK binding will be described in detail 
+here. If you would like to use another binding, you will need to follow the 
+README files in their respective repositories 
+([JikesRVM](https://github.com/mmtk/mmtk-jikesrvm), 
+[V8](https://github.com/mmtk/mmtk-v8))
+ to set them up, and find appropriate benchmarks for testing. 
+ Also, while it may be useful to fork the relevant repositories to your own 
+ account, it is not required for this tutorial.
+
+First, set up OpenJDK, MMTk, and the binding:
+1. Clone the [OpenJDK binding](https://github.com/mmtk/mmtk-openjdk).
+2. Clone the mmtk-core repository and the 
+[OpenJDK VM repository](https://github.com/mmtk/openjdk). 
+Place them both in `mmtk-openjdk/repos`.
+4. Ensure you can build OpenJDK according to the instructions in the READMEs of 
+[the mmtk-core repository](/../master/README.md) and the 
+[OpenJDK binding repository](https://github.com/mmtk/mmtk-openjdk/blob/master/README.md).
+   * Use the `slowdebug` option when building the OpenJDK binding. This is the 
+   fastest debug variant to build, and allows for easier debugging and better 
+   testing. The rest of the tutorial will assume you are using `slowdebug`.
+
+
+
+#### Test the build
+A few benchmarks of varying size will be used throughout the tutorial. If you 
+haven't already, set them up now. All of the following commands should be 
+entered in `repos/openjdk`.
+1. **HelloWorld** (simplest, will never trigger GC): 
+   1. Copy the following code into a new Java file titled "HelloWorld.java" 
+   in `mmtk-openjdk/repos/openjdk`:
+   ```java
+   class HelloWorld {
+       public static void main(String[] args) {
+           System.out.println("Hello World!");
+       }
+   }
+   ```
+   2. Use the command 
+   `./build/linux-x86_64-normal-server-$DEBUG_LEVEL/jdk/bin/javac HelloWorld.java`.
+   3. Then, run 
+   `./build/linux-x86_64-normal-server-$DEBUG_LEVEL/jdk/bin/java HelloWorld -XX:+UseThirdPartyHeap` 
+   to run HelloWorld.
+   
```

Comment:
If your program printed out `Hello World!` as expected, then congratulations, you have MMTk working with OpenJDK.